### ISSUE TITLE
[feature request] publish a arm64 docker image

### ISSUE THREAD

**Issue by hojerst** [DannyBen/bashly#164]
Hi,

I really like the bashly framework for creating small scripts - great work! As I'm developing on a M1 Mac: Would it be possible to provide a arm64 based Docker image as well? I was willing to create a pull request, however it seems the images are publish manually by you - at least I haven't found any code for it.

Here is a simple command that can be used to build a multi-arch image with docker and publish it directly:

```bash
# on first run only
docker buildx create --use

# build and publish the image with two architectures in one go
docker buildx build . --platform "linux/amd64,linux/arm64" --push -t name/image:tag
``` 

Prerequisites: baseimage is supporting arm64

**Comment by DannyBen** [comment]
Well, the docker image is provided as a convenience really, and yes - I am building the image locally and pushing - originally it was auto-built on Docker Hub, but they removed this feature from their community offering.

I ran the commands, and it failed with:

```
 => CANCELED [linux/amd64 3/3] RUN gem install bashly --version 0.7.1      0.3s
 => ERROR [linux/arm64 3/3] RUN gem install bashly --version 0.7.1         0.3s
------
 > [linux/arm64 3/3] RUN gem install bashly --version 0.7.1:
#10 0.131 .buildkit_qemu_emulator: /bin/sh: Invalid ELF image for this architecture
------
Dockerfile:6
--------------------
   4 |     WORKDIR /app
   5 |
   6 | >>> RUN gem install bashly --version 0.7.1
   7 |
   8 |     ENTRYPOINT ["bashly"]
--------------------
error: failed to solve: process "/dev/.buildkit_qemu_emulator /bin/sh -c gem install bashly --version 0.7.1" did not complete successfully: exit code: 255
```

The script I am using to build is here:
https://github.com/DannyBen/bashly/blob/1e313a0aba86aff4ce68b96bf880a859323c6a59/Runfile#L49-L54

**Comment by hojerst** [comment]
Seems your qemu system is not initialized correctly, so it doesn't really support running arm64. You can fix this with

```
docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
```

However keep in mind, that your base image also needs to support arm64 for this to work. I build a personal a copy of your base image and of bashly and pushed it to my own dockerhub account. However I assume it is better if you provide them at the official locations. (had to change the Dockerfile of bashly to use my baseimage instead, obviously - otherwise no changes where needed).

**Comment by DannyBen** [comment]
Thanks. Providing multi-arch docker images is not a priority for bashly. As I said, docker image is provided as a curtesy, not as an objective. If this would have been simpler to implement and test, I might have agreed to investigate further, but as things stand, at least at this point in time, I am unlikely to.
